#feature/1

StanislavDargel opened the pull request:
1. Product.java added new fields,generated new getters and setters, generated equals() and hashCode(), toString(). Information output there changed.
2. ShoppingListApplication.java fixed several lines, added new functions.
3. Created new class ProductCategory.java.
4. Created new class ProductRepository.java. ShoppingList dataBase was moved from ShoppingListApplication.java to ProductRepository.java.
5. Created new class ProductValidationExceptions.java.
6. Created new class ProductValidationService.java.
7. Created new class ValidationService.java.

StanislavDargel committed:
#feature/1
1. Product.java added new fields,generated new getters and setters, generated equals() and hashCode(), toString(). Information output there changed.
2. ShoppingListApplication.java fixed several lines, added new functions.
3. Created new class ProductCategory.java.
4. Created new class ProductRepository.java. ShoppingList dataBase was moved from ShoppingListApplication.java to ProductRepository.java.
5. Created new class ProductValidationExceptions.java.
6. Created new class ProductValidationService.java.
7. Created new class ValidationService.java.

javagururt commented on src/com/javaguru/shoppinglist/Product.java:
```diff
@@ -32,4 +69,53 @@ public void setPrice(BigDecimal price) {
+
+    @Override
+    public String toString() {
+        if (description != null && !description.isEmpty() && discount != null) {
```
неочевидно, что в toString будет логика

javagururt commented on src/com/javaguru/shoppinglist/ProductCategory.java:
```diff
@@ -0,0 +1,34 @@
+    public static void printProductCategory() {
+        int categorySequence = 1;
+
+        for(ProductCategory category: ProductCategory.values()) {
```
можно использовать ordinal

javagururt commented on src/com/javaguru/shoppinglist/ProductValidationService.java:
```diff
@@ -0,0 +1,85 @@
+    }
+
+    private void productNameNotEmpty(Product product) {
+        if (product.getName() == null || product.getName().isEmpty())
```
лучше использовать фигурные скобки

javagururt commented on src/com/javaguru/shoppinglist/ShoppingListApplication.java:
```diff
@@ -1,45 +1,146 @@
+                        }
+                        switch (categoryNum) {
+                            case 1:
+                                product.setCategory(ProductCategory.FRUIT);
```
можно написать метод, который вернет категорию в зависимости от переданного числа

javagururt reviewed (commented):
отлично

StanislavDargel committed:
#feature/1
1. Product.java. Method toString() reworked.
2. ProductCategory.java. Method printProductCategory() fixed.
3. ProductValidationExceptions.java. In methods braces was placed in the right places.
4. ShoppingListApplication.java reworked.
The functional part is divided into methods
